src/main/java/com/cloudant/client/internal/HierarchicalUriComponents.java

```diff
@@ -136,10 +136,21 @@ public boolean isAllowed(int c) {
         QUERY_PARAM {
             @Override
             public boolean isAllowed(int c) {
-                if ('=' == c || '+' == c || '&' == c) {
+                // The query component defined in RFC 3986 does not specify the structure of data
+                // within a query or query parameter, which is deferred to the URI's scheme.
+                // The HTTP scheme in turn does not specify any restrictions on the structure of
+                // data within the query component, only that it must conform to RFC 3986.
+                // As a result the valid characters are determined by the server processing the
+                // request, so in this case we defer to the conventions of
+                // application/x-www-form-urlencoded types forcing encoding of the following:
```

Comment:
This comment is nice and detailed, but is there a reference for `x-www-form-urlencoded` anywhere, is it a standard or convention?